### ISSUE TITLE
GMP is no more resource in PHP >=5.6

### ISSUE THREAD

**Issue by macmon** [rlanvin/php-ip#2]
On PHP 5.6 GMP is no more a resource object: http://php.net/manual/en/intro.gmp.php

This Code
```php
$ip = new IPv4Block("192.168.0.1/16");
```
throws in PHP 5.6 an Exception:
```
Fatal error: Uncaught exception 'InvalidArgumentException' with message 'Unsupported argument type: object' in ./vendor/rlanvin/php-ip/src/IPv4.php:90
Stack trace:
#0 ./vendor/rlanvin/php-ip/src/IPBlock.php(62): IPv4->__construct(Object(GMP))
#1 ./vendor/rlanvin/php-ip/src/IPBlock.php(127): IPBlock->getMask()
#2 ./test.php(5): IPBlock->__construct('192.168.0.1/16')
#3 {main}
  thrown in ./vendor/rlanvin/php-ip/src/IPv4.php on line 90
```

**Comment by rlanvin** [comment]
Thanks for letting me know! I do not have a php 5.6 to test right now, but provided nothing else has changed with GMP in PHP 5.6, I believe this can be fixed by modifying IPv4.php on line 83 and IPv6.php on line 82 to:

    elseif ( (is_resource($ip) && get_resource_type($ip) == 'GMP integer') || $ip instanceof GMP ) {

If you want to try, let me know if it works, and I'll commit (or make a pull request).

**Comment by macmon** [comment]
Yea, the fix work with PHP 5.4 and PHP 5.6. Thx.
